File: teledonJFX/src/main/java/com/example/teledonjfx/utils/events/DonationEvent.java
package com.example.teledonjfx.utils.events;

import com.example.teledonjfx.model.CharitableCase;

public class DonationEvent implements Event{
    private ChangeEventType type;
    private CharitableCase newCharitableCase, oldCharitableCase;

    public DonationEvent(ChangeEventType type, CharitableCase newCharitableCase) {
        this.type = type;
        this.newCharitableCase = newCharitableCase;
    }

    public DonationEvent(ChangeEventType type, CharitableCase newCharitableCase, CharitableCase oldCharitableCase) {
        this.type = type;
        this.newCharitableCase = newCharitableCase;
        this.oldCharitableCase = oldCharitableCase;
    }

    public ChangeEventType getType() {
        return type;
    }

    public CharitableCase getNewCharitableCase() {
        return newCharitableCase;
    }

    public CharitableCase getOldCharitableCase() {
        return oldCharitableCase;
    }
}

File: teledonDesktop/teledonDesktop/utils/observer/events/DonationEvent.cs
using teledonCS.model;

namespace teledonCS.utils.observer.events
{
    public class DonationEvent: Event
    {
        private ChangeEventType type;
        private CharitableCase charitableCase, oldCharitableCase;

        public DonationEvent(ChangeEventType type, CharitableCase charitableCase, CharitableCase oldCharitableCase)
        {
            this.type = type;
            this.charitableCase = charitableCase;
            this.oldCharitableCase = oldCharitableCase;
        }

        public DonationEvent(ChangeEventType type, CharitableCase charitableCase)
        {
            this.type = type;
            this.charitableCase = charitableCase;
        }

        public ChangeEventType getType()
        {
            return type;
        }

        public CharitableCase getCharitableCase()
        {
            return charitableCase;
        }

        public CharitableCase GetOldCharitableCase()
        {
            return oldCharitableCase;
        }
    }
}